test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -19,18 +18,21 @@ function ($, Q, Expect, $sandbox, DeviceFingerprint) {
       window.postMessage(JSON.stringify(message), '*');
     }
 
-    function mockUserAgent(userAgent) {
-      navigator = new Object(); // eslint-disable-line no-global-assign
-      navigator.__proto__ || (navigator.__proto__ = _navigator);
-      navigator.__defineGetter__('userAgent', function () { return userAgent; });
+    function mockUserAgentCheck(userAgent) {
```

Comment:
I think we can leave this as `mockUserAgent` since it is still doing that for the test purposes